Source/CA_library.cpp

```diff
@@ -582,73 +582,70 @@ void CellularAutomata::twodim_rule3(int k, int kprime)
 // Note: Kassady created this function but had trouble pushing it to the repo
 void CellularAutomata::update()
 {
-    // Create a temporary grid to store the updated state
     std::vector<std::vector<int>> temp_grid(rows, std::vector<int>(cols, 0));
 
-    // Iterate over each cell in the grid
     for (int i = 0; i < rows; ++i)
     {
         for (int j = 0; j < cols; ++j)
         {
-            // Get the states of the Von Neumann neighbors
             int north_state = grid[(i - 1 + rows) % rows][j];
             int south_state = grid[(i + 1) % rows][j];
             int east_state = grid[i][(j + 1) % cols];
             int west_state = grid[i][(j - 1 + cols) % cols];
 
-            // Use the determine_genotype() function to get the new state
-            int new_state = determine_genotype(grid[i][j], north_state, south_state, east_state, west_state);
+            // Call determine_genotype for each neighbor pair and decide the new state
+            // This is a simplification where we just take an average of the neighbors' influence
+            int sum_states = determine_genotype(north_state, south_state) + determine_genotype(east_state, west_state);
+            int new_state = round(static_cast<double>(sum_states) / 2.0);
 
-            // Update the temporary grid with the new state
             temp_grid[i][j] = new_state;
         }
     }
 
-    // Update the main grid with the new states from the temporary grid
-    grid = temp_grid;
+    grid.swap(temp_grid); // Efficient way to update the main grid
 }
 
 // This function is a specific rules function for our allele model of which
 // we were told to just include in the CA general purpose library.
-int CellularAutomata::determine_genotype(int cell_state, int north_state, int south_state, int east_state, int west_state)
+int CellularAutomata::determine_genotype(int cell_state1, int cell_state2)
```

Comment:
Determine_genotype() seems to have the two parameter version which is the correct one!